rust/pst/contract/implementation/src/contract.rs

```diff
@@ -1,14 +1,26 @@
 use async_recursion::async_recursion;
-
-use warp_pst::action::{Action, ActionResult};
-use warp_pst::state::State;
-
-use crate::actions::{Actionable, *};
-use warp_wasm_utils::contract_utils::js_imports::{Block, Contract, log, SmartWeave, Transaction};
-
+use warp_pst::{
+    action::{Action, ViewResponse},
+    error::ContractError,
+    state::State,
+};
+
+use crate::actions::*;
+use warp_wasm_utils::{
+    contract_utils::{
+        handler_result::ActionResult,
+        js_imports::{SmartWeave, Transaction},
+    },
+    warp_bindgen,
+};
+
+#[warp_bindgen]
 #[async_recursion(?Send)]
-pub async fn handle(state: State, action: Action) -> ActionResult {
-    let original_caller = Transaction::owner();
+pub async fn handle2(
+    state: State,
+    action: Action,
+) -> ActionResult<State, ViewResponse, ContractError> {
+    let _original_caller = Transaction::owner();
```

Comment:
what's that for? all functions use effective caller...effectively ;-)

Reply:
It was here already. I'll get rid of this.